### PR TITLE
Feat/hng 66 develop job listings page with pagination

### LINKED ISSUE #66: [Feature]: Develop Job Listings Page with Pagination
### Description

Develop a job listings page that fetches all job listings from the appropriate API endpoints, implements pagination to manage the display of listings, and uses the job listing card component to display each job. Ensure the page matches the provided design specifications and is responsive. The page will use the job listing card component created by another intern to present the job data.

### Acceptance Criteria

- [ ]  Fetch all job listings from the appropriate API endpoints
- [ ]  Implement pagination to manage the display of a large number of job listings.
- [ ]  Use the job listing card component created by another intern to display each job.
- [ ] Display empty states and not found state components where needed.
- [ ]  Ensure the job listings page matches the provided design specifications and is responsive
- [ ] Ensure smooth integration with existing components on the careers page.

**Deadline: Saturday 12Noon**
**Check #60 and #62 for the needed components**

### Links

https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=417-23224&m=dev
image

image

### PR DESCRIPTION
**Pull request by shuaibu-shehu**
# Job Listings Page Implementation Checklist

- [x] Fetch all job listings from the appropriate API endpoints
- [x] Implement pagination to manage the display of a large number of job listings
- [ ] Use the job listing card component created by another intern to display each job
- [x] Display empty states and not found state components where needed
- [x] Ensure the job listings page matches the provided design specifications and is responsive
- [x] Ensure smooth integration with existing components on the careers page

close #66 

[Screencast from 2024-07-20 15-51-52.webm](https://github.com/user-attachments/assets/85561405-55fe-4681-93b5-3d3362748260)

Screenshot from 2024-07-20 15-50-49

### CODE REVIEW

**Comment by shuaibu-shehu** [comment]
I could not get the needed compoents to include in the page

File: app/components/pagination-item.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import { Job } from "~/routes/career";
+import { Button } from "./ui/button";
+import { ChevronLeft, ChevronRight, MoreHorizontal } from "lucide-react";
```
Comment (chumex412) [inline app/components/pagination-item.tsx:3]: This pagination logic is not correct, please use ShadCN pagination component

File: app/routes/career.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        jobTitle: getRandomElement(jobTitles),
+        location: getRandomElement(locations),
+        description: getRandomElement(descriptions),
+        salaryRange: getRandomElement(salaryRanges),
+        jobType: getRandomElement(jobTypes),
+        jobCategory: getRandomElement(jobCategories),
+        detailsLink: `/jobs/${index}`
+    };
+};
+
```
Comment (chumex412) [inline app/routes/career.tsx:45]: You were told to use clean code, right? Why are these logics here? I only expected UI rendering logic in this file.

**Review by chumex412 — CHANGES_REQUESTED** [review]

**Comment by chumex412** [comment]
The implementation doesn't match the design.

**Comment by chumex412** [comment]
Also resolve the conflicts

File: app/components/pagination-item.tsx (hunk 1 above)
Comment (shuaibu-shehu) [reply app/components/pagination-item.tsx:3]: someone implemented the pagination already https://github.com/hngprojects/hng_boilerplate_remix/pull/214

base on my  task description i thought it is my responsibility

File: app/routes/career.tsx (hunk 2 above)
Comment (shuaibu-shehu) [reply app/routes/career.tsx:45]: ok, working in it sir

Comment (shuaibu-shehu) [reply app/routes/career.tsx:45]: @chumex412  but what do i do while most of the components need to display on the page are not available

**Comment by Jomagene** [comment]
I think you have to use the paginatio component I had to create.
@shuaibu-shehu

**Comment by shuaibu-shehu** [comment]
> I think you have to use the paginatio component I had to create. @shuaibu-shehu

@Jomagene  yaeh, is your "PR merged

File: app/routes/career.tsx (hunk 2 above)
Comment (chumex412) [reply app/routes/career.tsx:45]: Move the data and logic to a .ts file outside the .tsx file and import it into the .tsx file

Comment (Jomagene) [reply app/routes/career.tsx:45]: @chumex412 have you niticed our issue? Can I be merged so that @shuaibu-shehu may use the pegination I have implemented?

Comment (Jomagene) [reply app/routes/career.tsx:45]: see here https://github.com/hngprojects/hng_boilerplate_remix/pull/194#issuecomment-2241349278

Comment (shuaibu-shehu) [reply app/routes/career.tsx]: @chumex412  in could not find the figma design anymore

Comment (Jomagene) [reply app/routes/career.tsx]: @shuaibu-shehu How is it my brother?

**Pull request closed without merging**